doc/BUILDSERVER.md

```diff
@@ -0,0 +1,170 @@
+* jenkins build [libelektra](https://build.libelektra.org/jenkins/job/libelektra/) please
+* jenkins build [local-installation](https://build.libelektra.org/job/elektra-local-installation/) please
+* jenkins build [mingw64](https://build.libelektra.org/job/elektra-gcc-configure-mingw-w64/) please
+* jenkins build [multiconfig-gcc-stable](https://build.libelektra.org/job/elektra-multiconfig-gcc-stable/) please
+* jenkins build [source-package-test](https://build.libelektra.org/job/elektra-source-package-test/) please
+
+Additionally `jenkins build all please` can be used to trigger all build jobs
+relevant for PR's.
+Since this needs a lot of resources please use it only if
+
+- all of the **standard PR jobs** were already **successful**, and
+- you are sure that you **do not want change anything** in your PR anymore
+
+.
+
+## Test Environments
+We use Docker containers to provide the various test environments.
+
+They are described
+[in the repository](https://master.libelektra.org/scripts/docker)
+and the Jenkinsfile describes how to build them.
+If a rebuild of the images is needed is determined by the hash of the
+Dockerfile used to describe it.
+If it has not changed the build step will be skipped.
+In the case that an image needed a build it will afterwards be uploaded into a
+private Docker image registry (currently on a7) and thus is shared between all
+Docker capable build slaves.
+
+## Tests
+We will use the Docker images build as described earlier to run compilations
+and tests for Elektra.
+This allows us to run tests independent of which nodes are available (as the
+environment is portable).
+
+The Jenkinsfile describes the steps used to run tests.
+Helper functions for easily adding new tests are available
+(buildAndTest, BuildAndTestAsan, ...).
+
+Coverage reports are generated automatically when using the buildAndTest helper
+and the appropriate Cmake flags for coverage generation have been set. They are
+uploaded to https://doc.libelektra.org/coverage/.
+
+Artifacts from `ctest` are also preserved automatically when using
+buildAndTest.
+The function also takes care of providing a stagename based path so multiple tests
+can not overwrite files that share the same name.
+
+Tests are executed in order dictated by the Jenkinsfile.
+In general new tests should be added to the 'full build stage' that will only
+run after a standard full test run succeeded.
+This saves execution time on the build server for most common errors.
+
+Since there is no strict way to enforce the way tests are added we encourage
+you to read the existing configuration and modify existing tests so they suite
+your needs.
+
+## Deployment
+For runs of the build job that are run in the master branch we also execute
+deployment steps after all tests pass.
+We use it to build debian packages and move it into the repository on the a7
+node.
+Additionally we recompile the homepage and deploy it on the a7 node.
```

Comment:
still open?

Reply:
I added information re test artefacts + coverage reports and specified that homepage + debian packages are dployed onto a7. 
Do you want more detail?

Reply:
Again, I do not see it. Did you forget to push it or is there a problem in GitHub?

I only see:
```
+For runs of the build job that are run in the master branch we also execute
+deployment steps after all tests pass.
+We use it to build debian packages and move it into the repository on the a7
+node.
+Additionally we recompile the homepage and deploy it on the a7 node.
```